packages/flutter/lib/src/material/toggle_buttons.dart

```diff
@@ -30,8 +30,35 @@ import 'toggle_buttons_theme.dart';
 /// correlated by their index in the list. The length of [isSelected] has to
 /// match the length of the [children] list.
 ///
+/// There is a Material 3 version of this component, [SegmentedButton],
+/// that's preferred for applications that are configured for Material 3
+/// (see [ThemeData.useMaterial3]).
+///
 /// {@youtube 560 315 https://www.youtube.com/watch?v=kVEguaQWGAY}
 ///
+/// ## Updating to [SegmentedButton]
+///
+/// There is a Material 3 version of this component, [SegmentedButton],
+/// that's preferred for applications that are configured for Material 3
+/// (see [ThemeData.useMaterial3]). The [SegmentedButton] widget's visuals
+/// are a little bit different, see the Material 3 spec at
+/// <https://m3.material.io/components/segmented-buttons/overview> for
+/// more details. The [SegmentedButton] widget's API is also slightly different.
+/// While the [ToggleButtons] widget can have list of widgets, the
+/// [SegmentedButton] widget has a list of [ButtonSegment]s with
+/// a type value. While the [ToggleButtons] uses a list of boolean values
```

Comment:
I realize that by "type value" you mean a value of type `T` however getting `ButtonSegment<T>` into the API doc as a link is going to be messy, so I guess this is OK.

Reply:
Yes